smithy-build/src/main/java/software/amazon/smithy/build/transforms/FlattenNamespaces.java

```diff
@@ -0,0 +1,181 @@
+ */
+public final class FlattenNamespaces extends ConfigurableProjectionTransformer<FlattenNamespaces.Config> {
+
+    /**
+     * {@code removeTraitShapes} configuration settings.
+     */
+    public static final class Config {
+
+        private String namespace;
+        private ShapeId service;
+        private Set<String> tags = Collections.emptySet();
+
+        /**
+         * Sets the target namespace that existing namespaces will be flattened
+         * into.
+         *
+         * @param namespace The target namespace to use in the model.
+         */
+        public void setNamespace(String namespace) {
+            this.namespace = namespace;
+        }
+
+        /**
+         * Gets the target namespace that existing namespaces will be flattened
+         * into.
+         *
+         * @return the target namespace to be used in the model.
+         */
+        public String getNamespace() {
+            return namespace;
+        }
+
+        /**
+         * Sets the service ShapeId that will be flattened into the target
+         * namespace.
+         *
+         * @param service The ID of the service.
+         */
+        public void setService(ShapeId service) {
+            this.service = service;
+        }
+
+        /**
+         * @return Gets the service shape ID of the service that will have
+         * its shape namespaces updated.
+         */
+        public ShapeId getService() {
+            return service;
+        }
+
+        /**
+         * Sets the set of tags that are retained in the model.
+         *
+         * @param tags The tags to retain in the model.
+         */
+        public void setIncludeTagged(Set<String> tags) {
+            this.tags = tags;
+        }
+
+        /**
+         * Gets the set of tags that are retained in the model.
+         *
+         * @return Returns the tags to retain.
+         */
+        public Set<String> getIncludeTagged() {
+            return tags;
+        }
+    }
+
+    @Override
+    public Class<Config> getConfigType() {
+        return Config.class;
+    }
+
+    @Override
+    protected Model transformWithConfig(TransformContext context, Config config) {
+        if (config.getService() == null || config.getNamespace() == null) {
+            throw new SmithyBuildException(
+                    "'namespace' and 'service'properties must be set on flattenNamespace transformer.");
+        }
+        Model model = context.getModel();
+        Map<ShapeId, ShapeId> shapesToRename = getRenamedShapes(config, model);
+        return ModelTransformer.create().renameShapes(model, shapesToRename);
+    }
+
+    @Override
+    public String getName() {
+        return "flattenNamespaces";
+    }
+
+    private Map<ShapeId, ShapeId> getRenamedShapes(Config config, Model model) {
+        if (!model.getShape(config.getService()).isPresent()) {
+            throw new SmithyBuildException("Service not found in model when performing flattenNamespaces transform.");
```

Comment:
We should probably validate that the shape *is* a service too. We might end up broadening support of this transform in the future to support other shapes, but may need different logic.

Reply:
Added validation, with a separate error message, that this is needs to be a service shape.